Unable to compile USB console example with uart_mux

**Describe the bug**
Unable to compile USB console example with uart_mux enabled. Compiler complaining repeated variable declaration by  [cdc_acm.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/usb/class/cdc_acm.c#L1104) & [uart_mux.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/console/uart_mux.c#L881)

```
zephyrproject/zephyr/drivers/console/uart_mux.c:897: multiple definition of `rx_ringbuf_0'; 
zephyr/libzephyr.a(cdc_acm.c.obj):C:/Users/yongc/zephyrproject/zephyr/subsys/usb/class/cdc_acm.c:1127: first defined here

zephyrproject/zephyr/drivers/console/uart_mux.c:897: multiple definition of `tx_ringbuf_0'; 
zephyr/libzephyr.a(cdc_acm.c.obj):C:/Users/yongc/zephyrproject/zephyr/subsys/usb/class/cdc_acm.c:1127: first defined here
```

**To Reproduce**
Steps to reproduce the behavior:
1. Build [usb/console](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/subsys/usb/console) with `CONFIG_GSM_MUX` set
2. See compilation error

**Expected behavior**
Should compile fine (?)

**Impact**
Unable to use usb/console with gsm_mux

**Logs and console output**
Full compilation log: [usb_compilation_log.txt](https://github.com/zephyrproject-rtos/zephyr/files/6729988/usb_compilation_log.txt)

**Environment (please complete the following information):**
 - OS: Windows 10
 - Toolchain: gnuarmemb
 - Zephyr 2.6 RC2
 - stm32g0b1re custom board

**Additional context**
1. Using offline WIP usb driver
2. Can be workaround in the meantime by renaming either file's `&t(r)x_ringbuf_##x` to `&yyy_t(r)x_ringbuf_##x`

--- a/subsys/usb/class/cdc_acm.c
+++ b/subsys/usb/class/cdc_acm.c
@@ -1078,14 +1078,14 @@ static const struct uart_driver_api cdc_acm_driver_api = {
 		.endpoint = cdc_acm_ep_data_##x,			\
 	};								\
 									\
-	RING_BUF_DECLARE(rx_ringbuf_##x,				\
+	RING_BUF_DECLARE(cdc_acm_rx_rb_##x,				\
 			 CONFIG_USB_CDC_ACM_RINGBUF_SIZE);		\
-	RING_BUF_DECLARE(tx_ringbuf_##x,				\
+	RING_BUF_DECLARE(cdc_acm_tx_rb_##x,				\
 			 CONFIG_USB_CDC_ACM_RINGBUF_SIZE);		\
 	static struct cdc_acm_dev_data_t cdc_acm_dev_data_##x = {	\
 		.line_coding = CDC_ACM_DEFAULT_BAUDRATE,		\
-		.rx_ringbuf = &rx_ringbuf_##x,				\
-		.tx_ringbuf = &tx_ringbuf_##x,				\
+		.rx_ringbuf = &cdc_acm_rx_rb_##x,			\
+		.tx_ringbuf = &cdc_acm_tx_rb_##x,			\
 	};
 
 #define DT_DRV_COMPAT zephyr_cdc_acm_uart

